src/main/java/school/hei/haapi/endpoint/rest/mapper/UserMapper.java

```diff
@@ -90,12 +93,34 @@ public User toDomain(Student student) {
         .lastName(student.getLastName())
         .email(student.getEmail())
         .ref(student.getRef())
-        .status(User.Status.fromValue(student.getStatus().toString()))
+        .status(toDomainUserStatus(student.getStatus()))
         .phone(student.getPhone())
         .entranceDatetime(student.getEntranceDatetime())
         .birthDate(student.getBirthDate())
         .sex(student.getSex() == null ? null : User.Sex.valueOf(student.getSex().toString()))
         .address(student.getAddress())
         .build();
   }
+
+  private User.Status toDomainUserStatus(EnableStatus status) {
+    List<EnableStatus> expectedStatus = Arrays.stream(EnableStatus.values()).toList();
+    switch (status) {
+      case ENABLED:
+        return User.Status.ENABLED;
+      case DISABLED:
+        return User.Status.DISABLED;
+      default: throw new BadRequestException("Status must be type of: " + expectedStatus.toString());
+    }
+  }
+
+  private EnableStatus toRestUserStatus(User.Status status) {
+    List<User.Status> expectedStatus = Arrays.stream(User.Status.values()).toList();
```

Comment:
no need to do this because our actual app's behaviour on unexpected status is to throw a BadRequest with the given status